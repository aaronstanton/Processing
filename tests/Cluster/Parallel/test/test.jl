using Seismic

println("Started julia on Parallel.westgrid.ca")

# initialize only 1 process per node
Seismic.bind_torque_procs(1)

# here a function that runs your estimation:
# using MOpt, mig

# require some code on all nodes
require("incl.jl")

println("make everybody say hello")

@everywhere sayhello()

println("make everybody do some math")

pmap( i->domath(i), [100 for j in 1:length(workers())] )

println("make everybody pass a memory test")

pmap( i->doBIGmath(), 1:length(workers()) )

println("trying parallel for loop with $(nprocs()) processes")
println("numworkers: $(length(workers()))")
println("workers: $(workers())")
@time map( n -> sum(svd(rand(n,n))[1]) , [800 for i in 1:20]);
@time pmap( n -> sum(svd(rand(n,n))[1]) , [800 for i in 1:24]);

println(" quitting ")

quit()
