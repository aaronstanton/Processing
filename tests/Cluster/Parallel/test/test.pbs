#!/bin/bash 
#PBS -S /bin/bash
#PBS -l nodes=3:ppn=12 
#PBS -l mem=66gb 
#PBS -l walltime=0:30:00 
#PBS -m bea 
#PBS -M aaron.stanton@ualberta.ca
#PBS -N test
#PBS -o test.out
#PBS -e test.err

date

# Change to directory from which job was submitted
cd $PBS_O_WORKDIR 

echo "here is your PBS_NODEFILE"
cat $PBS_NODEFILE

echo "calling julia now"
julia test.jl

