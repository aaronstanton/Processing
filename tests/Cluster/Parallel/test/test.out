Mon Aug 17 11:49:23 MDT 2015
here is your PBS_NODEFILE
cn0757
cn0757
cn0757
cn0757
cn0757
cn0757
cn0757
cn0757
cn0757
cn0757
cn0757
cn0757
cn0798
cn0798
cn0798
cn0798
cn0798
cn0798
cn0798
cn0798
cn0798
cn0798
cn0798
cn0798
cn0805
cn0805
cn0805
cn0805
cn0805
cn0805
cn0805
cn0805
cn0805
cn0805
cn0805
cn0805
calling julia now
Started julia on Parallel.westgrid.ca
name of compute nodes and number of workers:
["cn0805"=>12,"cn0757"=>12,"cn0798"=>12]
added 1 processes on master itself
added 1 processes on machine cn0805
added 1 processes on machine cn0798
done
make everybody say hello
hi I am worker number 1, I live on cn0757
	From worker 2:	hi I am worker number 2, I live on cn0757
	From worker 3:	hi I am worker number 3, I live on cn0805
	From worker 4:	hi I am worker number 4, I live on cn0798
make everybody do some math
	From worker 2:	Hi, I am worker number 2 doing some math
	From worker 3:	Hi, I am worker number 3 doing some math
	From worker 4:	Hi, I am worker number 4 doing some math
make everybody pass a memory test
	From worker 2:	allocating a 30000 by 10000 matrix 
	From worker 3:	allocating a 30000 by 10000 matrix 
	From worker 4:	allocating a 30000 by 10000 matrix 
	From worker 3:	my matrix has 2.23 GB
	From worker 4:	my matrix has 2.23 GB
	From worker 2:	my matrix has 2.23 GB
trying parallel for loop with 4 processes
numworkers: 3
workers: [2,3,4]
elapsed time: 16.957320031 seconds (828984408 bytes allocated, 3.21% gc time)
elapsed time: 6.879629952 seconds (323236 bytes allocated)
 quitting 
